doc/build-windows.md

```diff
@@ -66,12 +69,29 @@ See also: [dependencies.md](dependencies.md).
 
 ## Building for 64-bit Windows
 
-To build executables for Windows 64-bit, install the following dependencies:
+The first step is to install the mingw-w64 cross-compilation tool chain. Due to different Ubuntu
+packages for each distribution and problems with the Xenial packages the steps for each are different.
+
+Ubuntu Trusty 14.04:
 
     sudo apt-get install g++-mingw-w64-x86-64 mingw-w64-x86-64-dev
 
-Then build using:
+Ubuntu Xenial 16.04 and Windows Subsystem for Linux ([^1][^2]):
+
+    sudo apt-get install g++-mingw-w64-x86-64 mingw-w64-x86-64-dev software-properties-common
+    sudo add-apt-repository "deb http://archive.ubuntu.com/ubuntu zesty universe"
+    sudo apt update
+    sudo apt upgrade
+    sudo update-alternatives --config x86_64-w64-mingw32-g++ # Set the default mingw32 g++ compiler option to posix.
+
+Ubuntu Zesty 17.04 ([^2]):
+
+    sudo apt-get install g++-mingw-w64-x86-64 mingw-w64-x86-64-dev
```

Comment:
µnit: This installation step is the same for all three version. Might as well mention it only once, mention that no further steps are needed on trusty and that other versions require some workarounds: ...